lib/adapters/TapeAdapter.js

```diff
@@ -0,0 +1,134 @@
+const EventEmitter = require('events');
+
+module.exports = class TapeAdapter extends EventEmitter {
+  /**
+   * Unlike other adapters, this adapter ends up disabling the default
+   * reporter of its framework (in this case, Tape's TAP reporter to
+   * console.log). This is the default Tape behaviour, and we choose not to
+   * change that. If you wish to enable Tape's default reporter also, and
+   * e.g. use CRI only for additional reports to network or artefact files,
+   * then run `tape.createStream().pipe(process.stdout);` to have both.
+   *
+   * @param {Tape} tape
+   */
+  constructor (tape) {
+    super();
+
+    // TODO: Unable to observe 'prerun' event for some reason.
+    let started = false;
+    let runStartTime;
+    let runStatus = 'passed';
+    const runCounts = { passed: 0, failed: 0, skipped: 0, todo: 0, total: 0 };
+    const startOnce = () => {
+      if (!started) {
+        started = true;
+        this.emit('runStart', { name: null, counts: { total: null } });
+        runStartTime = new Date().getTime();
+      }
+    };
+
+    let fakeTestId = -1;
+    const activeTests = {};
+    const handleRow = (row) => {
+      let test;
+      switch (row.type) {
+        // {type: test, name: '…', id: 0, skip: false, todo: false}
+        // {type: test, name: '…', id: 1, skip: false, todo: false, parent: 0}
+        case 'test':
+          test = activeTests[row.id] = {
+            startTime: new Date().getTime(),
```

Comment:
```suggestion
            startTime: Date.now(),
```
same throughout; since 2011 `new Date().getTime()` has been a legacy pattern :-)